packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1292,6 +1293,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     ): N.TsImportEqualsDeclaration {
       node.isExport = isExport || false;
       node.id = this.parseIdentifier();
+      this.checkLVal(
+        node.id,
+        BIND_LEXICAL,
```

Comment:
I'm not sure about which kind of binding this should be: https://www.typescriptlang.org/play/?esModuleInterop=false&target=99&module=1&ssl=3&ssc=17&pln=4&pc=17#code/JYWwDg9gTgLgBAQzgXjlApgRwK7AwCgCIFCBKAbgChKYBPMdRFOAO2xACN0oqAbdeAgBcrdlyjMATFUroAHpFhwA3kwC+5IA

Reply:
The binding type should be as same as import default specifier: https://www.typescriptlang.org/play/?esModuleInterop=false&target=99&module=1&ssl=1&ssc=1&pln=3&pc=1#code/JYWwDg9gTgLgBAQzgXjlApgRwK7AwCgCIYBPMdAZwGMpgwZCBKAbgChXTzEU4A7bEACN0UNgBt08BAC4+A4VB4AmNq3QAPSLDgBvbgF9mQA

The declaration conflict error is shadowed by the module-not-found.